internal/util/pkgutil/pkgutil.go

```diff
@@ -15,25 +15,67 @@
 package pkgutil
 
 import (
-	"fmt"
+	"io"
 	"io/ioutil"
 	"os"
 	"path/filepath"
+	"sort"
 	"strings"
 
 	kptfilev1alpha2 "github.com/GoogleContainerTools/kpt/pkg/api/kptfile/v1alpha2"
 	"github.com/GoogleContainerTools/kpt/pkg/kptfile/kptfileutil"
 	"sigs.k8s.io/kustomize/kyaml/copyutil"
 )
 
-type RemoteSubPkgInfo struct {
-	PackagePath          string
-	DeclaringKptfilePath string
+// FindAllDirectSubpackages returns the paths to all immediate subpackages of the
+// provided path. It finds both remote and local subpackages, but it will not
+// find any nested packages. The path to the root package is not returned.
+func FindAllDirectSubpackages(path string) ([]string, error) {
```

Comment:
I think we have a method for returning direct packages with new pkg abstraction. See if you can use that ?

Reply:
Yeah, but there are some issues with using it. It is currently tied the package abstraction, which will require some larger changes to incorporate into the packaging code. It also isn't actually correct, since it only looks for subpackages in the immediate subdirectories. Trying to reconcile the different functions for traversing package hierarchies is on my list.

Reply:
Can you add TODOs for future refactoring here

Reply:
Done. Some of this work is already in https://github.com/GoogleContainerTools/kpt/pull/1583